File: C++/Hard/WrongAnswer/car-fleet-ii_0.cpp
class Solution {
public:
    vector<double> getCollisionTimes(vector<vector<int>>& cars) {
        
        int n = cars.size();
        vector<double> res(n, -1);
        stack<int> st;
        for (int i = n - 1; i >= 0; --i) {
            int p = cars[i][0], s = cars[i][1];
            while (!st.empty()) {
                int j = st.top();
                int p2 = cars[j][0], s2 = cars[j][1];
                if ((s2 >= s) || (res[j] > 0 && (p2 - p) / (s - s2) >= res[j])) {
                    st.pop();
                } else {
                    break;
                }
            }
            if (!st.empty()) {
                int j = st.top();
                int p2 = cars[j][0], s2 = cars[j][1];
                res[i] = (p2 - p) / (s - s2);
            }
            st.push(i);
        }
        return res;
    }
};

File: Rust/Hard/CompileError/car-fleet-ii_0.rs
impl Solution {
    pub fn get_collision_times(cars: Vec<Vec<i32>>) -> Vec<f64> {
        
        let mut stack = Vec::new();
        let mut res = vec![-1.0; cars.len()];
        for i in (0..cars.len()).rev() {
            let (pos, speed) = (cars[i][0] as f64, cars[i][1] as f64);
            while let Some(&j) = stack.last() {
                let (npos, nspeed) = (cars[j][0] as f64, cars[j][1] as f64);
                if speed <= nspeed || (res[j] > 0.0 && (npos - pos) / (speed - nspeed) >= res[j]) {
                    stack.pop();
                } else {
                    break;
                }
            }
            if let Some(&j) = stack.last() {
                let (npos, nspeed) = (cars[j][0] as f64, cars[j][1] as f64);
                res[i] = (npos - pos) / (speed - nspeed);
            }
            stack.push(i);
        }
        res
    }
}